static/js/controllers/contacts.js

```diff
@@ -95,9 +96,17 @@ var SP_ID_MASK = 0xFFFFF8;
               // move it to the top
               contacts.splice(homeIndex, 1);
               contacts.unshift(usersHomePlace);
-            } else if (!$scope.filters.search) {
+            } else if (!$scope.filters.search && !$scope.filters.simprintsIdentities) {
               contacts.unshift(usersHomePlace);
             }
+            if ($scope.filters.simprintsIdentities) {
+              contacts.forEach(function(contact) {
+                var identity = $scope.filters.simprintsIdentities.find(function(identity) {
+                  return identity.id === contact.simprints_id;
```

Comment:
Is there a reason to return the `identify()` result as a json object `{ id: id }`, rather than just a string?